text/2952-futures.md

```diff
@@ -59,52 +59,35 @@
 
 自從最初的 futures 0.3 發布以來，除了上面提到的改進之外，核心 `Future` 特徵和任務系統的變化相對較小。實際的 `Future` 特徵基本上與 4 月時相同。
 
-# Guide-level explanation
-[guide-level-explanation]: #guide-level-explanation
+# 教學式解說
+[教學式解說]: #guide-level-explanation
 
-The `Future` trait represents an *asynchronous* and lazy computation that may
-eventually produce a final value, but doesn't have to block the current thread
-to do so.
-
-Futures can be constructed through `async` blocks or `async` functions, e.g.,
+`Future` 特徵代表了一個非同步和惰性計算，它最終可能會產生一個最終值，但不以阻塞當前執行緒為前提來達成。
 
+Future 可以通過 `async` 區塊或 `async` 函式來創建，例如
 ```rust
 async fn read_frame(socket: &TcpStream) -> Result<Frame, io::Error> { ... }
 ```
 
 This `async` function, when invoked, produces a future that represents the
 completion of reading a frame from the given socket. The function signature
 is equivalent to:
+這個 `async` 函式，當被呼叫時，產生一個 future，代表從给定的 socket 中讀取一個幀的完成。函式簽名等同於：
 
 ```rust
 fn read_frame<'sock>(socket: &'sock TcpStream)
     -> impl Future<Output = Result<Frame, io::Error>> + 'sock;
 ```
 
-Other async functions can *await* this future; see the [companion
-RFC] for full details.
-
-In addition to `async fn` definitions, futures can be built using adapters, much
-like with `Iterator`s. Initially these adapters will be provided entirely "out
-of tree", but eventually they will make their way into the standard library.
-
-Ultimately asynchronous computations are executed in the form of *tasks*,
-which are comparable to lightweight threads. *executor*s provide the ability to
-create tasks from `()`-producing `Future`s. The executor will pin the `Future`
-and `poll` it until completion inside the task that it creates for it.
-
-The implementation of an executor schedules the tasks it owns in a cooperative
-fashion. It is up to the implementation of an executor whether one or more
-operation system threads are used for this, as well as how many tasks can be
-spawned on it in parallel. Some executor implementations may only be able to
-drive a single `Future` to completion, while others can provide the ability to
-dynamically accept new `Future`s that are driven to completion inside tasks.
-
-This RFC does not include any definition of an executor. It merely defines the
-interaction between executors, tasks and `Future`s, in the form of APIs
-that allow tasks to request getting scheduled again.
-The `task` module provides these APIs, which are required when manually implementing
-`Future`s or executors.
+其他非同步函數可以 **await** 這個 future; 有關完整詳細信息，請參閱[隨附的 RFC](https://github.com/rust-lang/rfcs/pull/2394)。
+
+除了 `async fn` 定義，future 還可以使用適配器來創建，就像 `Iterator` 一樣。最初，這些適配器將完全由「out of tree」提供，但最终它們將帶入標準函式庫。
```

Comment:
I'm not sure what is "tree" to mean in here
I think this means the "other crates"

Reply:
An in-tree or out-of-tree module generally refers to whether a module is in the main codebase or not. 

In rust, we usually see modules or functions residing rust-lang/rust without using git subtree/submodules as in-tree, otherwise is out-of-tree.

Reply:
I current have no better term to translate out-of-tree neither.